code/game/machinery/OpTable.dm

```diff
@@ -48,19 +48,19 @@
 		return FALSE
 
 
-/obj/machinery/optable/MouseDrop_T(atom/movable/O as mob|obj, var/mob/P)
-	if(usr.stat || (!ishuman(P) && !isrobot(P)) || !check_table())
+/obj/machinery/optable/MouseDrop_T(atom/movable/O, mob/user)
+	if(!ishuman(user) && !isrobot(user)) //Only Humanoids and Cyborgs can put things on this table
 		return
-	if(!check_table()) //Is the Operating Table currently occupied?
+	if(!check_table()) //Is the Operating Table empty?
 		return
-	if(P.restrained() || P.buckled || P.incapacitated()) //Folks who are restrained or otherwise incapacitated cannot get on the table
+	if(user.restrained() || user.buckled || user.IsWeakened() || user.stunned || user.incapacitated()) //Is the person trying to use the table incapacitated or restrained?
 		return
-	if(!ismob(O)) //Humans only
+	if(!ishuman(O)) //Only Humanoids can go on this table
 		return
-	if(istype(O, /mob/living/simple_animal) || istype(O, /mob/living/silicon)) //animals and robots dont fit
+	if(isanimal(O) || isrobot(O)) //Animals and Cyborgs do not go on the table
 		return
 	var/mob/living/L = O
```

Comment:
```suggestion
	var/mob/living/carbon/C = O
```
Both because you've checked for this above, and because `take_patient` accepts a `mob/living/carbon`, and not just  any `mob/living`.

But also
```dm
if(!iscarbon(O))
  return
var/mob/living/carbon/C = O
```
is the same as
```dm
var/mob/living/carbon/C = O
if(!C)
  return
```

I'm leaning a bit towards the latter, because it's harder to screw up.

Consider the following snippet (roughly what used to be here):
```dm
if(!ismob(O))
  return
var/mob/living/L = O
```
Now `L` can suddenly be null, and you might not notice (because you've checked that it's a mob already! Why would you check it again, right?)! (The previous code tried to work around this by snowflaking simple animals and silicons, but those "fixes" only hide the problem, and don't actually resolve it).

Reply:
Good catch! Thank you!

Reply:
```
	if(!ishuman(user) && !isrobot(user)) //Only Humanoids and Cyborgs can put things on this table
		return
	if(!check_table()) //Is the Operating Table empty?
		return
	if(user.restrained() || user.buckled || user.IsWeakened() || user.stunned || user.incapacitated()) //Is the person trying to use the table incapacitated or restrained?
		return
	var/mob/living/carbon/C = O
	if(!C) //Only Carbon mobs can go on this table, if C is null, get out of here
		return
	take_patient(C, user)
```